compiler/aarch64/codegen/ControlFlowEvaluator.cpp

```diff
@@ -128,6 +128,15 @@ if (cg->profiledPointersRequireRelocation() && secondChild->getOpCodeValue() ==
    }
 #endif
 
+   int64_t secondChildValue = is64bit ? secondChild->getLongInt() : secondChild->getInt();
+   if ((TR::CC_EQ == cc || TR::CC_NE == cc)
+         && 0 == secondChildValue)
+      {
+      TR::LabelSymbol *dstLabel = node->getBranchDestination()->getNode()->getLabel();
+      TR::Instruction *result = generateCbzOrCbnzInstruction(cg, node, src1Reg, dstLabel, is64bit);
+      return result;
```

Comment:
You have to take care of the following:
- Call `decReferenceCount()` on the child nodes before returning from this function.
- The parent node may have the third child node for `TR::GlRegDeps`.  You need to call `generateCompareBranchInstruction()` with `TR::RegisterDependencyConditions` for the third child in that case.

Reply:
- I can see the   `decReferenceCount()` on the child nodes, at bottom before returning results. I will bring a copy of those call here, or will avoid returning the result from this point.
- For the third child, `generateConditionalBranchInstruction()` is called in the code. Will it be `generateCompareBranchInstruction` or `generateConditionalBranchInstruction()`?

Reply:
You cannot generate `cbz/cbnz` instructions with `generateConditionalBranchInstruction()`.

Please use `generateCompareBranchInstruction()` for `cbz/cbnz`.
```
generateCompareBranchInstruction(cg, TR::InstOpCode::cbzx, node, src1Reg, dstLabel);
```
When the node has the third child, you can call the function with `deps` like this:
```
generateCompareBranchInstruction(cg, TR::InstOpCode::cbzx, node, src1Reg, dstLabel, deps);
```

Reply:
Made these changes.